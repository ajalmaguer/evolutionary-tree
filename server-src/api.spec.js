const requestPromise = require('request-promise');

const baseUrl = 'http://localhost:4200/api';

const rp = (method, url, body) => {
    return requestPromise(baseUrl + url, {
        method: method,
        body: body,
        json: true,
    });
};

describe('#getNodes', () => {
    test('returns all nodes with amoeba as root node', () => {
        return rp('GET', '/nodes')
            .then(res => expect(res).toMatchSnapshot());
    });
});

describe('#getNodesById', () => {
    test('returns child node and all its children', () => {
        const id = '5c9c27c1ab34d91ecd77027d';
        return rp('GET', '/nodes/' + id)
            .then(res => {
                expect(res.rootNode.id).toEqual(id);
                expect(res).toMatchSnapshot();
            });
    });
});

// rp('POST', '/nodes').then(res => console.log('res =', res))
// rp('GET', '/nodes').then(res => console.log('res =', res))
// rp('GET', '/nodes/123').then(res => console.log('res =', res))
// rp('PUT', '/nodes/123').then(res => console.log('res =', res))
// rp('DELETE', '/nodes/123').then(res => console.log('res =', res))