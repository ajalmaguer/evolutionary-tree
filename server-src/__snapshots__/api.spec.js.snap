// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#getNodes returns all nodes with amoeba as root node 1`] = `
Object {
  "nodesById": Object {
    "5c9c27c1ab34d91ecd77027b": Object {
      "__v": 1,
      "_id": "5c9c27c1ab34d91ecd77027b",
      "childIds": Array [
        "5c9c27c1ab34d91ecd77027c",
        "5c9c27c1ab34d91ecd77027d",
      ],
      "id": "5c9c27c1ab34d91ecd77027b",
      "name": "Amoeba",
      "path": null,
    },
    "5c9c27c1ab34d91ecd77027c": Object {
      "__v": 0,
      "_id": "5c9c27c1ab34d91ecd77027c",
      "childIds": Array [],
      "id": "5c9c27c1ab34d91ecd77027c",
      "name": "Virus",
      "path": ",5c9c27c1ab34d91ecd77027b,",
    },
    "5c9c27c1ab34d91ecd77027d": Object {
      "__v": 1,
      "_id": "5c9c27c1ab34d91ecd77027d",
      "childIds": Array [
        "5c9c27c1ab34d91ecd77027e",
      ],
      "id": "5c9c27c1ab34d91ecd77027d",
      "name": "Bacteria",
      "path": ",5c9c27c1ab34d91ecd77027b,",
    },
    "5c9c27c1ab34d91ecd77027e": Object {
      "__v": 0,
      "_id": "5c9c27c1ab34d91ecd77027e",
      "childIds": Array [],
      "id": "5c9c27c1ab34d91ecd77027e",
      "name": "Algae",
      "path": ",5c9c27c1ab34d91ecd77027b,5c9c27c1ab34d91ecd77027d,",
    },
  },
  "rootNode": Object {
    "__v": 1,
    "_id": "5c9c27c1ab34d91ecd77027b",
    "childIds": Array [
      "5c9c27c1ab34d91ecd77027c",
      "5c9c27c1ab34d91ecd77027d",
    ],
    "id": "5c9c27c1ab34d91ecd77027b",
    "name": "Amoeba",
    "path": null,
  },
}
`;

exports[`#getNodesById returns child node and all its children 1`] = `
Object {
  "nodesById": Object {
    "5c9c27c1ab34d91ecd77027d": Object {
      "__v": 1,
      "_id": "5c9c27c1ab34d91ecd77027d",
      "childIds": Array [
        "5c9c27c1ab34d91ecd77027e",
      ],
      "id": "5c9c27c1ab34d91ecd77027d",
      "name": "Bacteria",
      "path": ",5c9c27c1ab34d91ecd77027b,",
    },
    "5c9c27c1ab34d91ecd77027e": Object {
      "__v": 0,
      "_id": "5c9c27c1ab34d91ecd77027e",
      "childIds": Array [],
      "id": "5c9c27c1ab34d91ecd77027e",
      "name": "Algae",
      "path": ",5c9c27c1ab34d91ecd77027b,5c9c27c1ab34d91ecd77027d,",
    },
  },
  "rootNode": Object {
    "__v": 1,
    "_id": "5c9c27c1ab34d91ecd77027d",
    "childIds": Array [
      "5c9c27c1ab34d91ecd77027e",
    ],
    "id": "5c9c27c1ab34d91ecd77027d",
    "name": "Bacteria",
    "path": ",5c9c27c1ab34d91ecd77027b,",
  },
}
`;
